### DISCUSSION TITLE
Lesson 6 - forge not work and not installed

### DISCUSSION THREAD

**Discussion by Soheilprs** [Cyfrin/foundry-full-course-cu#374] [Q&A] [answered]
Hi everyone, i recently install ubuntu and the "foundryup" command install everything except the "foundry" and when i enter the "foundry init" or any foundry command i receive this error:
```
FORGE - training program for SNAP (version 2006-07-28)

usage: forge [options] <ann> <dna> [options]
options:
  -help
  -verbose
  -pseudocount <float>  [1]   (absolute number for all models)
  -pseudoCoding <float> [0.0] (eg. 0.05)
  -pseudoIntron <float> [0.0]
  -pseudoInter <float>  [0.0]
  -min-counts           [0]
  -lcmask [-fragmentN]
  -utr5-length <int>    [50]
  -utr5-offset <int>    [10]
  -utr3-length <int>    [50]
  -utr3-offset <int>    [10]
  -explicit <int>       [250]
  -min-intron <int>     [30]
  -boost <file>  (file of ID <int>)

```

**Comment by ChefAharoni** [comment]
The correct command is `forge init`.

**Reply by Soheilprs** [reply]
When i enter this i also receive this error :)

**Reply by ChefAharoni** [reply]
When you installed foundry through `foundryup`, did you get a message that foundry was installed successfully?

**Reply by Soheilprs** [reply]
Yes its says it want to install forge, chisel, anvil and ... but at the end it install everything except forge

**Reply by ChefAharoni** [reply]
Try building it directly from the source:
```bash
cargo install --git https://github.com/foundry-rs/foundry --profile local --force foundry-cli anvil chisel
```

Make sure you have [Rust](https://www.rust-lang.org/tools/install) installed.

If not, you can install it like stated in the link above:
```bash
curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
```

[Link to direction in Foundry Book](https://book.getfoundry.sh/getting-started/installation#building-from-source)

**Reply by Soheilprs** [reply]
I will test it and let you know, thanks

**Reply by AztecWarlord** [reply]
try closing the terminal then reload it.

**Reply by Soheilprs** [reply]
.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx
 
 ╔═╗ ╔═╗ ╦ ╦ ╔╗╔ ╔╦╗ ╦═╗ ╦ ╦         Portable and modular toolkit
 ╠╣  ║ ║ ║ ║ ║║║  ║║ ╠╦╝ ╚╦╝    for Ethereum Application Development 
 ╚   ╚═╝ ╚═╝ ╝╚╝ ═╩╝ ╩╚═  ╩                 written in Rust.

.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx

Repo       : https://github.com/foundry-rs/
Book       : https://book.getfoundry.sh/                      
Chat       : https://t.me/foundry_rs/                         
Support    : https://t.me/foundry_support/
Contribute : https://github.com/orgs/foundry-rs/projects/2/

.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx

curl: (7) Failed to connect to 127.0.0.1 port 2080 after 0 ms: Connection refused
foundryup: command failed: curl -sSf https://api.github.com/repos/foundry-rs/foundry/git/refs/tags/nightly
foundryup: installing foundry (version nightly, tag nightly-)
foundryup: downloading latest forge, cast, anvil, and chisel
curl: (7) Failed to connect to 127.0.0.1 port 2080 after 0 ms: Connection refused

foundryup: command failed: download https://github.com/foundry-rs/foundry/releases/download/nightly-/foundry_nightly_linux_amd64.tar.gz

gzip: stdin: unexpected end of file
tar: Child returned status 1
tar: Error is not recoverable: exiting now
foundryup: command failed: tar -xzC /home/soheil/.foundry/bin
foundryup: downloading manpages
curl: (7) Failed to connect to 127.0.0.1 port 2080 after 0 ms: Connection refused


gzip: stdin: unexpected end of file
tar: Child returned status 1
tar: Error is not recoverable: exiting now
foundryup: installed - forge 0.2.0 (0e72b71 2023-07-19T00:31:45.646186071Z)
foundryup: warning: 
There are multiple binaries with the name 'forge' present in your 'PATH'.
This may be the result of installing 'forge' using another method,
like Cargo or other package managers.
You may need to run 'rm /usr/bin/forge' or move '/home/soheil/.foundry/bin'
in your 'PATH' to allow the newly installed version to take precedence!

foundryup: installed - cast 0.2.0 (0e72b71 2023-07-19T00:31:45.646186071Z)
foundryup: installed - anvil 0.1.0 (0e72b71 2023-07-19T00:32:33.293158643Z)
foundryup: installed - chisel 0.1.0 (0e72b71 2023-07-19T00:32:33.296881269Z)
foundryup: done!

**Reply by ChefAharoni** [reply, marked as answer]
> ```There are multiple binaries with the name 'forge' present in your 'PATH'.
> This may be the result of installing 'forge' using another method,
> like Cargo or other package managers.
> You may need to run 'rm /usr/bin/forge' or move '/home/soheil/.foundry/bin'
> in your 'PATH' to allow the newly installed version to take precedence! ```

**This** is your problem. It looks like you have two packages of *forge* in your PATH. Try one of the commands stated in the error:
```bash
rm /usr/bin/forge
```
or 
```bash
mv /home/soheil/.foundry/bin /usr/bin/forge
```

**Reply by Soheilprs** [reply]
thanks, 
works well

**Reply by ChefAharoni** [reply]
Great, I'm glad to hear.

Please mark my previous reply as an answer, to let others know this issue is solved.
